### PR TITLE
Upgrading libraries

### PR DESCRIPTION
**Pull request by javierfernandes**
Major changes, but not in terms of functionality but it terms of dependencies and tools

* Upgraded react to 16
* Upgraded redux and all related libraries
* Migrating from mocha+chai to jest
* Introduced ramda and refactor to reduce some code
* Increasing coverage to 100% :)
* Fixing dependencies (organizing). Almost all of them are just devDeps instead of direct deps

Greetings to my mom that is watching the show ! :)

### CODE REVIEW

File: src/core/model.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 }
 
 // checking value
-export const isFetchingSlot = slot => hasValueInKey(slot, 'state', State.FETCHING)
-export const isFetchedSlot = slot => hasValueInKey(slot, 'state', State.FETCHED)
-export const isErrorSlot = slot => hasValueInKey(slot, 'state', State.ERROR)
+const stateIs = value => obj => !!obj && obj.state === value
+export const isFetchingSlot = stateIs(State.FETCHING)
+export const isFetchedSlot = stateIs(State.FETCHED)
+export const isErrorSlot = stateIs(State.ERROR)
```
Comment (p4bloch) [inline src/core/model.js]: Maybe we can now remove the `Slot` suffix now that they don't receive arguments? Also, have you considered `isFetching`, `hasFetched` and `hasErrors`? They make more sense gramatically.

File: src/core/service.js [hunk 2]
```diff
@@ -52,10 +51,12 @@ function doCallEndpoint(callSpec, next, onReceive, onError, store) {
       // }
       (response.ok
         ? response.json().then(json => next(transformReceiveAction(onReceive((json)), store)))
+        // TODO: error parsing should be configurable globally (here we assume that a json object comes
+        // with { error: { message } }
         : response.json().then(json => next(onError(json.error.message))))
     )
     .catch(error => {
-      console.error('ERROR on API Call', error)
+      // console.error('ERROR on API Call', error)
```
Comment (p4bloch) [inline src/core/service.js]: What is the proper error handling solution we want to implement here? Should this throw instead? Probably out of scope of this PR though

File: src/core/actions.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
+import { flip, concat, propEq } from 'ramda'
```
Comment (nbarrera) [inline src/core/actions.js]: ❤️ Missing 🐏 da power 🚀

File: src/core/service.js (hunk 2 above)
Comment (javierfernandes) [reply src/core/service.js]: good question, this all runs async' as a dispatch. We should understand how to handle it. Maybe throwing and then from your componente you can do `.catch()` Still I believe that 95% of the times you don't want to handle the error on components. If it is for UI we already provide the state on the placeholder prop to show an error in the UI. So probably no body will care about this (?)

**Pull request merged**